packages/socket.io-client/lib/socket.ts

```diff
@@ -449,7 +449,10 @@ export class Socket<
       this.flags.volatile && (!isTransportWritable || !this.connected);
     if (discardPacket) {
       debug("discard packet as the transport is not currently writable");
-    } else if (this.connected) {
+    } else if (
+      this.connected &&
+      (this.flags.volatile || this.io.isResponsive())
```

Comment:
Is the `volatile` check needed here?

Reply:
I thought it made sense because doesn’t `volatile` mean you either want the packet sent immediately or discarded? So we could try and send it regardless if whether a heart beat has been missed because if it fails it doesn’t matter